File: library/src/main/java/nl/endevelopment/r2randroid/r2rlib/models/ExternalLink.java
package nl.endevelopment.r2randroid.r2rlib.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by jan on 11/07/16.
 */

public class ExternalLink implements Parcelable {
    /**
     text 	        string 	Link text
     url 	        string 	Link URL
     displayUrl 	string 	Display URL (cleaned up link url) (optional)
     moustacheUrl 	string 	Moustach URL (link url with replacement parameters) (optional)
     */

    private String text;
    private String url;
    private String displayUrl;
    private String moustacheUrl;

    public ExternalLink() {
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getDisplayUrl() {
        return displayUrl;
    }

    public void setDisplayUrl(String displayUrl) {
        this.displayUrl = displayUrl;
    }

    public String getMoustacheUrl() {
        return moustacheUrl;
    }

    public void setMoustacheUrl(String moustacheUrl) {
        this.moustacheUrl = moustacheUrl;
    }

    protected ExternalLink(Parcel in) {
        text = in.readString();
        url = in.readString();
        displayUrl = in.readString();
        moustacheUrl = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(text);
        dest.writeString(url);
        dest.writeString(displayUrl);
        dest.writeString(moustacheUrl);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<ExternalLink> CREATOR = new Parcelable.Creator<ExternalLink>() {
        @Override
        public ExternalLink createFromParcel(Parcel in) {
            return new ExternalLink(in);
        }

        @Override
        public ExternalLink[] newArray(int size) {
            return new ExternalLink[size];
        }
    };

    @Override
    public String toString() {
        return "ExternalLink{" +
                "text='" + text + '\'' +
                ", url='" + url + '\'' +
                ", displayUrl='" + displayUrl + '\'' +
                ", moustacheUrl='" + moustacheUrl + '\'' +
                '}';
    }
}

File: library/src/main/java/com/getroadmap/r2rlib/models/ExternalLink.kt
package com.getroadmap.r2rlib.models

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by jan on 11/07/16.
 */

open class ExternalLink() : Parcelable {
    /**
     * text 	        string 	Link text
     * url 	        string 	Link URL
     * displayUrl 	string 	Display URL (cleaned up link url) (optional)
     * moustacheUrl 	string 	Moustach URL (link url with replacement parameters) (optional)
     */

    var text: String? = null
    var url: String? = null
    var displayUrl: String? = null
    var moustacheUrl: String? = null

    constructor(parcel: Parcel) : this() {
        text = parcel.readString()
        url = parcel.readString()
        displayUrl = parcel.readString()
        moustacheUrl = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(text)
        parcel.writeString(url)
        parcel.writeString(displayUrl)
        parcel.writeString(moustacheUrl)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ExternalLink> {
        override fun createFromParcel(parcel: Parcel): ExternalLink {
            return ExternalLink(parcel)
        }

        override fun newArray(size: Int): Array<ExternalLink?> {
            return arrayOfNulls(size)
        }
    }

}